modules/javafx.controls/src/main/java/javafx/scene/control/DatePicker.java

```diff
@@ -147,11 +148,20 @@ public DatePicker(LocalDate localDate) {
 
         focusedProperty().addListener(o -> {
             if (!isFocused()) {
-                commitValue();
+                commitValueOnFocusLost();
```

Comment:
minor: I don't know if it's worth creating a new method here - try .. catch can simply be moved here.